examples/systemcalls/systemcalls.c

```diff
@@ -58,6 +67,35 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    // Fork the process to create a new child process
+    pid_t pid = fork();
+
+    // Check if fork was successful
+    if (pid == -1)
+    {
+        // Handle fork error
+        perror("fork");
+        exit(-1);
+    }
+    else if (pid == 0)
+    {
+        // Execute the specified command
+        execv(command[0], command);
+        // Handle execv error
+        perror("execv");
+        exit(-1);
+    }
+    else
+    {
+        int status;
+        // Wait for the child process to exit and get its exit status
+        waitpid(pid, &status, 0);
+
+        // Clean up variable argument list
+        va_end(args);
+        // Return true if the child process exited normally with status 0, otherwise false
+        return (WIFEXITED(status) && WEXITSTATUS(status) == 0) ? true : false;
```

Comment:
@Kanin-McGuire what happens to `va_end` cleanup in each of these paths?
Suggest restructuring so your function return isn't dead code.